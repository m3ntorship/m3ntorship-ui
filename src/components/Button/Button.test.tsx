import React from 'react';
import * as renderer from 'react-test-renderer';

import { Button } from './index';

describe('Button', () => {
  it('should render button component with {Default inner text, and className = {shared variant-filled-shared size-large variant-filled-primary}}', () => {
    const tree = renderer.create(<Button />);
    expect(tree).toMatchSnapshot();
  });
  it('should return ghost disabled button', () => {
    const tree = renderer.create(
      <Button content="secondary disabled" variant="ghost" disabled={true} />
    );
    expect(tree).toMatchSnapshot();
  });
});
