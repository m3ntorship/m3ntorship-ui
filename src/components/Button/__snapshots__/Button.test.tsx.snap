// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render button component with {Default inner text, and className = {shared variant-filled-shared size-large variant-filled-primary}} 1`] = `
<button
  className="shared variant-filled-shared size-md variant-filled-primary"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  Default
</button>
`;

exports[`Button should render loading spinner and invisible text when we set loading to true 1`] = `
<button
  className="shared variant-filled-shared size-md variant-filled-primary transparent"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  <div
    className="loading-wrapper"
  >
    <div
      className="loading"
    />
  </div>
  button text
</button>
`;

exports[`Button should return Button with stront tag child 1`] = `
<button
  className="shared variant-ghost-shared size-md variant-ghost-primary"
  disabled={true}
  onClick={[Function]}
  type="button"
>
  <strong>
     This is Test 
  </strong>
</button>
`;

exports[`Button should return ghost disabled button 1`] = `
<button
  className="shared variant-ghost-shared size-md variant-ghost-primary"
  disabled={true}
  onClick={[Function]}
  type="button"
>
  Disabled Button
</button>
`;
